starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const router = express.Router();
+
+const User = require('../models/users')
+
+// set up bcrypt
+const bcrypt = require('bcrypt')
+const bcryptSalt     = 10;
+
+
+// show form to signup
+router.get('/signup', function(req, res, next) {
+  res.render('auth/signup')
+});
+
+// process form signup
+router.post('/signup', (req, res, next) => {
+
+  const username = req.body.username;
+  const password = req.body.password;
+
+  if (username === "" || password === "") {
+    res.render("auth/signup", {
+      errorMessage: "Indicate a username and a password to sign up"
+    });
+    return;
+  }
+
+  User.findOne( {username: username}, (err, user) =>{
+    if (user !== null) {
+      res.render("auth/signup", {
+        errorMessage: "The username already exists"
+      });
+      return;
+    }
+
+    const salt     = bcrypt.genSaltSync(bcryptSalt);
+    const hashPass = bcrypt.hashSync(password, salt);
+
+    console.log('username: ', username)
+    console.log('password: ', hashPass)
+
+
+
+    var newUser  = new User({
+      username: username,
+      password: hashPass
+    });
+
+    newUser.save((err) => {
+      console.log('save')
+      res.redirect('/')
+    })
+  })
+})
+
+router.get('/login', (req, res, next) => {
+  res.render('auth/login')
+})
+
+router.post('/login', (req, res, next) => {
```

Comment:
you can not signup because you are not receiving the password value correctly. You are expecting a password param from the signup view, the problem is that your password input in the form has not value in the "name" attribute which is the responsible for giving the proper name to the paramateres